pull/github.go

```diff
@@ -149,30 +149,60 @@ func (ghc *GitHubContext) ChangedFiles() ([]*File, error) {
 
 func (ghc *GitHubContext) Commits() ([]*Commit, error) {
 	if ghc.commits == nil {
-		if err := ghc.loadPullRequestData(); err != nil {
-			return nil, err
+		totalCommits := ghc.pr.GetCommits()
+
+		var q struct {
+			Repository struct {
+				Object struct {
+					Commit struct {
+						History struct {
+							PageInfo v4PageInfo
+							Nodes    []*v4Commit
+						} `graphql:"history(first: $limit, after: $cursor)"`
+					} `graphql:"... on Commit"`
+				} `graphql:"object(oid: $oid)"`
+			} `graphql:"repository(owner: $owner, name: $name)"`
+		}
+		qvars := map[string]interface{}{
+			// always list commits from the head repository
+			// some commit details do not propagate from forks to the upstream
+			"owner": githubv4.String(ghc.pr.GetHead().GetRepo().GetOwner().GetLogin()),
+			"name":  githubv4.String(ghc.pr.GetHead().GetRepo().GetName()),
+			"oid":   githubv4.GitObjectID(ghc.pr.GetHead().GetSHA()),
+
+			"limit":  githubv4.Int(min(totalCommits, MaxPageSize)),
+			"cursor": (*githubv4.String)(nil),
 		}
-	}
 
-	if len(ghc.commits) >= MaxPullRequestCommits {
-		return nil, errors.Errorf("too many commits in pull request, maximum is %d", MaxPullRequestCommits)
-	}
+		var commits []*v4Commit
+		for len(commits) < totalCommits {
+			if err := ghc.v4client.Query(ghc.ctx, &q, qvars); err != nil {
+				return nil, errors.Wrap(err, "failed to list commits")
+			}
+			commits = append(commits, q.Repository.Object.Commit.History.Nodes...)
 
-	// verify that the head of the PR being evaluated exists in commit list
-	// some GitHub APIs have a delay propagating commit information
-	headSHA := ghc.pr.GetHead().GetSHA()
```

Comment:
I removed this check because we now list commits starting with the head SHA directly. If it doesn't exist, we'll get an empty response from the API call, which will either fail to deserialize or will fail on the commit count check.